components/prize_pool/wikis/rainbowsix/prize_pool_legacy.lua

```diff
@@ -0,0 +1,158 @@
+	local args = Template.retrieveReturnValues('LegacyPrizePool')
+	local header = Array.sub(args, 1, 1)[1]
+	local slots = Array.sub(args, 2)
+
+	local newArgs = {}
+
+	newArgs.prizesummary = (header.prizeinfo and not header.noprize) and true or false
+	newArgs.cutafter = header.cutafter
+
+	if Currency.raw(header.localcurrency) then
+		newArgs.localcurrency = header.localcurrency
+		inputToId.localprize = 'localprize'
+	else
+		LegacyPrizePool._assignType(newArgs, header, 'localcurrency', 'localprize')
+	end
+
+	LegacyPrizePool._assignType(newArgs, header, 'points')
+	LegacyPrizePool._assignType(newArgs, header, 'points2')
+	LegacyPrizePool._assignType(newArgs, header, 'points3')
+
+	if args.indiv then
+		newArgs.type = {type = 'solo'}
+	else
+		newArgs.type = {type = 'team'}
+	end
+
+	for slotIndex, slot in ipairs(slots) do
+		newArgs[slotIndex] = LegacyPrizePool._mapSlot(slot)
+	end
+	for link, idx in pairs(qualCache) do
+		newArgs['qualifies' .. idx] = link
+	end
+
+	return CustomPrizePool.run(newArgs)
+end
+
+function LegacyPrizePool._mapSlot(slot)
+	if not slot.place then
+		return {}
+	end
+
+	local newData = {}
+	if slot.place:lower() == 'dq' or slot.place:lower() == 'dnp' then
+		newData[slot.place:lower()] = true
+	else
+		newData.place = slot.place
+	end
+
+	newData.date = slot.date
+	newData.usdprize = (slot.usdprize and slot.usdprize ~= '0') and slot.usdprize or nil
+
+	Table.iter.forEachPair(inputToId, function(parameter, newParameter)
+		local input = slot[parameter]
+		if newParameter == 'seed' then
+			local links = LegacyPrizePool._parseWikiLink(input)
+			for _, link in ipairs(links) do
+				if not qualCache[link] then
+					qualCache[link] = nextQual
+					nextQual = nextQual + 1
+				end
+				newData['qualified' .. qualCache[link]] = true
+			end
+		elseif input and input ~= 0 then
+			newData[newParameter] = input
+		end
+	end)
+
+	for opponentIndex = 1, 128 do
+		if not slot[opponentIndex] then
+			break
+		end
+
+		local newOpponent = {}
+		newOpponent[1] = slot[opponentIndex]
+		newOpponent.date = slot['date' .. opponentIndex]
+		newOpponent.wdl = slot['wdl' .. opponentIndex]
+		newOpponent.lastvs = slot['lastvs' .. opponentIndex]
+		newOpponent.lastscore = slot['lastscore' .. opponentIndex]
+		newOpponent.lastvsscore = slot['lastvsscore' .. opponentIndex]
+
+		newData[opponentIndex] = newOpponent
+	end
+
+	return newData
+end
+
+function LegacyPrizePool._assignType(assignTo, args, parameter, slotParam)
+	slotParam = slotParam or parameter
+	local input = args[parameter]
+	if LegacyPrizePool._isValidPoints(input) then
+		assignTo['points' .. nextPoints] = input
+		inputToId[slotParam] = 'points' .. nextPoints
+		nextPoints = nextPoints + 1
+	elseif input == 'seed' then
+		inputToId[slotParam] = 'seed'
+	elseif String.isNotEmpty(input) then
+		assignTo['freetext' .. nextFreetext] = mw.getContentLanguage():ucfirst(input)
+		inputToId[slotParam] = 'freetext' .. nextFreetext
+		nextFreetext = nextFreetext + 1
+	end
+end
```

Comment:
probably also needs some checks for valid currency
the old `Template:Local currency` supports some currencies we do not want to support further (seed, percent, travel support, title, hardware, misc, points) hence they need to be catched imo

Reply:
This should already be handled

Reply:
ah yeah, me being blind there^^